system_metrics_collector/src/system_metrics_collector/linux_memory_measurement_node.cpp

```diff
@@ -124,4 +125,30 @@ double LinuxMemoryMeasurementNode::periodicMeasurement()
   return processMemInfoLines(read_string);
 }
 
+void LinuxMemoryMeasurementNode::publishStatistics()
+{
+  if (publisher_ == nullptr) {
+    return;
+  }
+
+  const moving_average_statistics::StatisticData statistic_data = getStatisticsResults();
+  const double data[moving_average_statistics::STATISTICS_DATA_TYPES.size()] = {
+    statistic_data.average,
+    statistic_data.min,
+    statistic_data.max,
+    statistic_data.standard_deviation,
+    static_cast<double>(statistic_data.sample_count)
+  };
+
+  MetricsMessage msg = newMetricsMessage();
+  msg.metrics_source = "memory_usage";
+  for (int i = 0; i < moving_average_statistics::STATISTICS_DATA_TYPES.size(); ++i) {
+    msg.statistics.emplace_back();
+    msg.statistics.back().data_type = moving_average_statistics::STATISTICS_DATA_TYPES[i];
+    msg.statistics.back().data = data[i];
+  }
```

Comment:
This code is a copy and paste of a logic somewhere else, please refactor to only have one code path.

Reply:
Done refactor. The functionality of this block of code now occurs only once, in `metrics_message_publisher.cpp`.